homeassistant/components/withings/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for the Withings API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/withings/
+"""
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.helpers import config_validation as cv
+
+from . import config_flow, const
+from .common import _LOGGER, create_withings_data_manger, NotAuthenticatedError
+from .sensor import WITHINGS_MEASUREMENTS_MAP
+
+DOMAIN = const.DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(const.CLIENT_ID): vol.All(cv.string, vol.Length(min=1)),
+                vol.Required(const.CLIENT_SECRET): vol.All(
+                    cv.string, vol.Length(min=1)
+                ),
+                vol.Optional(const.BASE_URL): cv.url,
+                vol.Required(const.PROFILES): vol.All(
+                    cv.ensure_list,
+                    vol.Unique(),
+                    vol.Length(min=1),
+                    [vol.All(cv.string, vol.Length(min=1))],
+                ),
+                vol.Optional(
+                    const.MEASURES, default=list(WITHINGS_MEASUREMENTS_MAP)
```

Comment:
All measurements should be created without an option to select them. See:
https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md